game/hud/src/widgets/Crafting/components/Input/index.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * @Author: Mehuge (mehuge@sorcerer.co.uk)
+ * @Date: 2017-05-14 21:42:18
+ * @Last Modified by: Mehuge (mehuge@sorcerer.co.uk)
+ * @Last Modified time: 2017-05-17 19:17:09
+ */
+
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { client } from 'camelot-unchained';
+
+interface InputProps {
+  size?: number;
+  value?: string;
+  disabled?: boolean;
+  onChange: (value: string) => void;
+}
+
+interface InputState {
+  value: string;
+  changed: boolean;
+}
+
+class Input extends React.Component<InputProps, InputState> {
+  constructor(props: InputProps) {
+    super(props);
+    this.state = { changed: false, value: props.value };
```

Comment:
You are being nitpicky :)  You are trading off valuable vertical screen space on a subjective notion of what constitutes readability.  I find this completely readable, whereas loosing vertical screen space decreases readability of code imo, because more scrolling is required.